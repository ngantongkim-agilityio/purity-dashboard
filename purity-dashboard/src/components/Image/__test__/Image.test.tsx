// Libs
import { render, fireEvent, act } from '@testing-library/react';

// Components
import { Image } from '..';

describe('Image component', () => {
  const mockProps = {
    width: 100,
    height: 100,
    alt: 'Test Image',
    src: '/test.src'
  };

  it('Should render Image component with default props', () => {
    const { container } = render(<Image {...mockProps} />);

    expect(container).toMatchSnapshot();
  });

  it('Should show fallbackSrc after loading get error', async () => {
    const { getByAltText } = render(<Image {...mockProps} />);
    const image = getByAltText('Test Image') as HTMLImageElement;

    await act(async () => {
      fireEvent.error?.(image);
    });

    expect(image.getAttribute('src')).toContain('empty-image.webp');
  });
});
