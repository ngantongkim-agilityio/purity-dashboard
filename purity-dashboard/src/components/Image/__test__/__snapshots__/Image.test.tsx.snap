// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component Should render Image component with default props 1`] = `
<div>
  <img
    alt="Test Image"
    class="transition-opacity object-cover duration-200 ease-in-out opacity-80"
    data-nimg="1"
    decoding="async"
    height="100"
    src="http://localhost/_next/image?url=%2Ftest.src&w=256&q=75"
    srcset="/_next/image?url=%2Ftest.src&w=128&q=75 1x, /_next/image?url=%2Ftest.src&w=256&q=75 2x"
    style="color: transparent;"
    width="100"
  />
</div>
`;
